runtime-modules/content/src/lib.rs

```diff
@@ -1418,13 +1440,30 @@ impl<T: Trait> Module<T> {
         assets: &StorageAssets<T>,
         channel_id: &T::ChannelId,
         prize_source_account: &T::AccountId,
+        channel_creation: bool,
     ) -> DispatchResult {
-        // construct upload params
-        let upload_params =
-            Self::pick_upload_parameters_from_assets(assets, channel_id, prize_source_account);
+        // dynamic bag for the channel
+        let dyn_bag = DynamicBagIdType::<T::MemberId, T::ChannelId>::Channel(*channel_id);
+        let bag_id = BagIdType::from(dyn_bag.clone());
+
+        let upload_params = UploadParametersRecord {
+            bag_id: bag_id.clone(),
+            object_creation_list: assets.object_creation_list.clone(),
+            deletion_prize_source_account_id: prize_source_account.clone(),
+            expected_data_size_fee: assets.expected_data_size_fee,
+        };
+
+        // create dynamic bag if it doesn't exist (does not emits conflicting created/deleted bag events)
+        if channel_creation && Storage::<T>::ensure_bag_exists(&bag_id).is_err() {
```

Comment:
Please, move this code to the `create_channel` extrinsic since it is the only place that needs it. It also will highlight the missing check for creation in that extrinsic.